src/runtime/components/forms/Input.vue

```diff
@@ -172,11 +177,40 @@ export default defineComponent({
       }
     }
 
-    const onInput = (event: InputEvent) => {
-      emit('update:modelValue', (event.target as HTMLInputElement).value)
+    // Custom function to handle the v-model properties
+    const updateInput = (value: string) => {
+
+      if (props.modelModifiers.trim) {
+        value = value.trim()
+      }
+
+      if (props.modelModifiers.number || props.type === 'number') {
+        value = looseToNumber(value)
+      }
```

Comment:
Why is this needed? The .number modifier seems to work correctly already:
https://stackblitz.com/edit/nuxt-ui-jaqqh1?file=app.vue

Reply:
I added it so its the same as with the v-model docs. You would need the if anyways for the props.type="number" as that doesn't work the same way as the native input right now: https://stackblitz.com/edit/nuxt-ui-pfqfnh?file=app.vue